timetypes/duration_type_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package timetypes_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+
+	"github.com/hashicorp/terraform-plugin-framework-timetypes/timetypes"
+)
+
+func TestDurationTypeValueFromTerraform(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		in          tftypes.Value
+		expectation attr.Value
+		expectedErr string
+	}{
+		"true": {
+			in:          tftypes.NewValue(tftypes.String, "72h3m0.5s"),
+			expectation: timetypes.NewDurationValueFromStringMust("72h3m0.5s"),
+		},
+		"unknown": {
+			in:          tftypes.NewValue(tftypes.String, tftypes.UnknownValue),
+			expectation: timetypes.NewDurationUnknown(),
+		},
+		"null": {
+			in:          tftypes.NewValue(tftypes.String, nil),
+			expectation: timetypes.NewDurationNull(),
+		},
+		"wrongType": {
+			in:          tftypes.NewValue(tftypes.Number, 123),
+			expectedErr: "can't unmarshal tftypes.Number into *string, expected string",
+		},
+	}
+	for name, testCase := range testCases {
+		name, testCase := name, testCase
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ctx := context.Background()
+
+			got, err := timetypes.DurationType{}.ValueFromTerraform(ctx, testCase.in)
+			if err != nil {
+				if testCase.expectedErr == "" {
+					t.Fatalf("Unexpected error: %s", err)
+				}
+				if testCase.expectedErr != err.Error() {
+					t.Fatalf("Expected error to be %q, got %q", testCase.expectedErr, err.Error())
+				}
+				return
```

Comment:
Nit: Given use of `(*testing.T).Fatalf()`, which will immediately stop the test, `return` is not necessary

```suggestion
```